modules/evm/src/runner/handler.rs

```diff
@@ -179,16 +179,11 @@ impl<'vicinity, 'config, 'meter, T: Config> Handler<'vicinity, 'config, 'meter,
 	}
 
 	fn transfer(transfer: Transfer) -> Result<(), ExitError> {
-		let source = T::AddressMapping::get_account_id(&transfer.source);
-		let target = T::AddressMapping::get_account_id(&transfer.target);
-
-		T::Currency::transfer(
-			&source,
-			&target,
-			transfer.value.saturated_into::<u128>().unique_saturated_into(),
-			ExistenceRequirement::AllowDeath,
-		)
-		.map_err(|_| ExitError::OutOfGas)
+		if transfer.value.is_zero() {
+			Ok(())
+		} else {
+			Err(ExitError::Other("invalid action".into()))
```

Comment:
Maybe should return `OutOfFund` instead. Basically no one has any EVM native token (which is not ACA)

Reply:
> https://github.com/AcalaNetwork/Acala/blob/958f6148038070415931089fdac389538024ee65/modules/evm/src/runner/mod.rs#L160-L168
> 
> This also needs update

This will not support `payable`. Is this the desired result？

Reply:
Yes. People should just use ERC20. Later we may want to convert all our build-in tokens to ERC677 to support similar semantic of payable.